esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -109,20 +109,20 @@ class CEsdlSDSStore : implements IEsdlStore, public CInterface
 
     virtual IPropertyTree* fetchBinding(const char* espProcess, const char* espStaticBinding) override
     {
-        VStringBuffer xpath("%s[@espprocess='%s'][@espbinding='%s'][1]", ESDL_BINDING_ENTRY, espProcess, espStaticBinding);
+        VStringBuffer xpath("%s[@espprocess='%s'][@espbinding='%s'][1]", ESDL_BINDING_PATH, espProcess, espStaticBinding);
         try
         {
-            Owned<IRemoteConnection> conn = querySDS().connect(ESDL_BINDINGS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
+            Owned<IRemoteConnection> conn = querySDS().connect(xpath.str(), myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
             if (!conn)
             {
-                ESPLOG(LogMin, "Unable to connect to ESDL Service binding information in dali %s", ESDL_BINDINGS_ROOT_PATH);
+                ESPLOG(LogMin, "Unable to connect to ESDL Service binding information in dali %s", xpath.str());
                 return nullptr;
             }
 
             ESPLOG(LogNormal, "ESDL Binding: Fetching ESDL Binding from Dali %s", xpath.str());
 
-            if (conn->queryRoot()->hasProp(xpath))
-                return createPTreeFromIPT(conn->queryRoot()->queryPropTree(xpath));
+            if (conn->queryRoot())
```

Comment:
@jakesmith can queryRoot() ever be non null?

Reply:
Not at the moment, but it really should be cleared by close().
Either way it is dangerous/undefined behaviour after close().